Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -120,14 +120,34 @@ void FullscreenManager::requestFullscreenForElement(Ref<Element>&& element, RefP
         return;
     }
 
+    // https://fullscreen.spec.whatwg.org/#fullscreen-element-ready-check
+    auto fullscreenElementReadyCheck = [checkType] (auto element, auto document) -> std::optional<ASCIILiteral> {
```

Comment:
ASCIILiteral has a null state and a boolean operator, you shouldn't need to use `std::optional`, just like for `String`.